### PR TITLE
Powershell for interface endpoints

### PR DESCRIPTION
**Pull request by chandrasekarsrinivasan**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by maddieclayton** [comment]
@chandrasekarsrinivasan Please either fill out or link a cmdlet design review here: https://github.com/Azure/azure-powershell-cmdlet-review-pr

**Comment by praries880** [comment]
design review : https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/152

**Comment by praries880** [comment]
fix CI issues :
* https://travis-ci.org/Azure/azure-powershell/builds/426415071#L1125
* https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/7307/console : 
  * 12:54:59   CSC : error CS2001: Source file 'c:\workspace\powershell\src\ResourceManager\Network\Commands.Network\InterfaceEndpoint\GetAzureServiceEndpointPolicyCommand.cs' could not be found. [c:\workspace\powershell\src\ResourceManager\Network\Commands.Network\Commands.Network.csproj]

**Comment by praries880** [comment]
@chandrasekarsrinivasan  you need to generate the help file for the cmdlet you added here.

**Comment by praries880** [comment]
@chandrasekarsrinivasan  kindly merge form the september branch to resolve conflicts

File: src/ResourceManager/Network/Commands.Network/ChangeLog.md [hunk 1]
```diff
@@ -45,6 +45,7 @@
 ## Version 6.7.0
 * Updated cmdlet New-AzureRmVirtualNetworkGatewayConnection with support for switch ExpressRouteGatewayBypass
 * Added cmdlets for Azure SdWan
+* Add cmdlet for Interface Endpoint Get-AzureInterfaceEndpoint
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ChangeLog.md]: this needs to be under `Current Release`

**Comment by praries880** [comment]
@chandrasekarsrinivasan  merge conflicts need to be resolved

File: src/ResourceManager/Profile/Commands.Profile/AzureRmAlias/Mappings.cs [hunk 2]
```diff
@@ -1045,7 +1045,8 @@ public class Mappings
                        'Get-AzApplicationGatewayUrlPathMapConfig':  'Get-AzureRmApplicationGatewayUrlPathMapConfig',
                        'Get-AzNetworkWatcherConnectionMonitor':  'Get-AzureRmNetworkWatcherConnectionMonitor',
                        'Remove-AzVirtualNetworkSubnetConfig':  'Remove-AzureRmVirtualNetworkSubnetConfig',
-                       'Get-AzBgpServiceCommunity':  'Get-AzureRmBgpServiceCommunity'
+                       'Get-AzBgpServiceCommunity':  'Get-AzureRmBgpServiceCommunity',
+                       'Get-AzInterfaceEndpoint':  'Get-AzInterfaceEndpoint',
```
Comment (praries880) [inline src/ResourceManager/Profile/Commands.Profile/AzureRmAlias/Mappings.cs]: there is something wrong here...
It should be 
'Get-AzInterfaceEndpoint':  'Get-AzureRmInterfaceEndpoint',

File: src/ResourceManager/Network/Commands.Network/InterfaceEndpoint/GetAzureInterfaceEndpointCommand.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+namespace Microsoft.Azure.Commands.Network
+{
+    using Microsoft.Azure.Commands.Network.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.WindowsAzure.Commands.Utilities.Common;
+    using System.Collections.Generic;
+    using System.Management.Automation;
+
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "InterfaceEndpoint", SupportsShouldProcess = true, DefaultParameterSetName = "GetByNameParameterSet"), OutputType(typeof(PSInterfaceEndpoint))]
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/InterfaceEndpoint/GetAzureInterfaceEndpointCommand.cs]: you need to adda  test case for this cmdlet. There is not test case in the PR

**Comment by praries880** [comment]
marking as `Do Not Merge` as the PR is missing attest case for the new cmdlet

File: src/ResourceManager/Network/Commands.Network/InterfaceEndpoint/GetAzureInterfaceEndpointCommand.cs (hunk 3 above)
Comment (chandrasekarsrinivasan) [reply src/ResourceManager/Network/Commands.Network/InterfaceEndpoint/GetAzureInterfaceEndpointCommand.cs]: Further details - we are not ready to expose end to end scenario for this through powershell. There is no way to set this on PS and hence a test case cannot be created. In a couple of months, we will have full support on this and then we can create  New-*, Set-* APIs.

**Comment by praries880** [comment]
@chandrasekarsrinivasan  kindly take care of merge conflicts

**Pull request merged**